lite/proxy/block.go

```diff
@@ -11,11 +11,17 @@ import (
 )
 
 func ValidateBlockMeta(meta *types.BlockMeta, check lite.Commit) error {
+	if meta == nil {
```

Comment:
do we have cases where we expect to receive nil blocks/metas?

Reply:
I was working on tests and a validation case would be to pass in nil. Here we crash. My goal is so that
dependencies of this package won't be surprised. We've had a couple of embarassing cases where there
were nil dereferences and panics from users.